cmd/xl-v1-healing-common.go

```diff
@@ -97,11 +101,37 @@ func listOnlineDisks(disks []StorageAPI, partsMetadata []xlMetaV1, errs []error)
 }
 
 // outDatedDisks - return disks which don't have the latest object (i.e xl.json).
-func outDatedDisks(disks, latestDisks []StorageAPI, partsMetadata []xlMetaV1, errs []error) (outDatedDisks []StorageAPI) {
+func outDatedDisks(disks, latestDisks []StorageAPI, partsMetadata []xlMetaV1,
+	bucket, object string) (outDatedDisks []StorageAPI) {
+
 	outDatedDisks = make([]StorageAPI, len(disks))
 	for index, disk := range latestDisks {
+		// disk either has an older xl.json or is not online.
 		if disk == nil {
 			outDatedDisks[index] = disks[index]
+			continue
+		}
+
+		// disk has a valid xl.json but may not have all the
+		// parts. This is considered an outdated disk, since
+		// it needs healing too.
+		for pIndex, part := range partsMetadata[index].Parts {
+			// compute blake2b sum of part.
+			hash := newHash(blake2bAlgo)
+			blakeBytes, hErr := hashSum(disk, bucket, filepath.Join(object, part.Name), hash)
+			if hErr != nil {
```

Comment:
a temporary network outage might cause hashSum to return error for all latest disks and hence cause healObject() to remove objects from all the latest disks.
To keep healing code simple, on encountering any kind of unexpected error we can just bailout and return error, the client can send the heal request again.